Source/Provers/SMTLib/SMTLibProcessTheoremProver.cs

```diff
@@ -1209,6 +1209,7 @@ protected SolverOutcome ParseOutcome(SExpr resp, out bool wasUnknown)
         case "error":
           if (resp.Arguments.Length == 1 && resp.Arguments[0].IsId &&
               (resp.Arguments[0].Name.Contains("max. resource limit exceeded")
+               || resp.Arguments[0].Name.Contains("push canceled")
```

Comment:
Is this because the only reason a push is ever cancelled is when Boogie itself determines resource limits are reached?

Reply:
Yes, exactly.